api/handlers/authentication.go

```diff
@@ -254,3 +257,145 @@ func sendOTP(userInfo string) {
 	}
 	fmt.Println("Email sent successfully!")
 }
+
+// resetPassword sends an email to the user with a link to reset their password
+// @Summary Reset a user's password
+// @Description Reset a user's password
+// @Tags auth
+// @Accept json
+// @Produce json
+// @Success 200 {object} models.Response "Email sent successfully"
+// @Failure 400 {object} models.Response "Invalid Request"
+// @Router /auth/reset-password/send-email [post]
+
+func (h Auth) ResetPassword(c *gin.Context) {
+	defer c.Request.Body.Close()
+	
+	//check the body of the request
+
+	var body models.SendResetRequest;
+	if err := c.BindJSON(&body); err != nil {
+		c.JSON(http.StatusBadRequest, models.Response{Error: "Invalid Request"})
+		return
+	}
+
+	//check if the email exists in the database
+
+	var user models.User;
+	h.DB.Where("email = ?", body.Email).First(&user);
+	if !h.checkUserExists(user.Email) {
+		c.JSON(http.StatusNotFound, models.Response{Error: "User does not exist"})
+		return
+	}
+
+
+	//generate a jwt token with the user's email
+	tempUser:= models.User{
+		Email: user.Email,
+	}
+
+	jwt, err := middleware.GenerateJWT(tempUser)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.Response{Error: "Error generating token"})
+		return
+	}
+
+	//send an email to the user with a temporary link to reset the password
+	linkURL := "http://localhost:8080/reset-password/" + jwt
```

Comment:
This should be link to the frontend. For dev environment this is `localhost:3000`, while for production it's the domain we got from DNS